.github/workflows/schedule-clear-cache.yml

```diff
@@ -0,0 +1,13 @@
+name: Clear cache weekly
+
+on:
+  schedule:
+    # on Sundays
+    - cron: "0 0 * * 0"
+
+jobs:
+  clear-cache:
+    if: ${{ github.repository_owner == 'Lightning-AI' }}
+    uses: Lightning-AI/utilities/.github/workflows/clear-cache.yml@main
+    with:
+      pattern: 'latest|docs'
```

Comment:
What does this do?

Reply:
The `'latest|docs'` will be used to grep the cache keys in the repository, and then, the workflow removes caches of grepped keys.

<details><summary>example caches to be cleared weekly</summary>

```bash
$ gh api -H "Accept: application/vnd.github+json" /repos/Lightning-AI/lightning/actions/caches --paginate --jq ".actions_caches[].key" | sort | uniq | grep -E 'latest|docs'
Linux-docs-make-pip-
Linux-docs-test-pip-
Linux-docs-test-pip-tdd19264-
Linux-pip-3.8-latest-912b4a7de1f6df12ab94aa2ec56c9614851f7662a0f1d8831f883e89ec11eba6
Linux-pip-3.8-latest-td2752-20d28281bf584b5b194e82c698718338e2c99ac9b6aa28e2ac6cc409b9d4c841
Linux-pip-3.8-latest-td2752-912b4a7de1f6df12ab94aa2ec56c9614851f7662a0f1d8831f883e89ec11eba6
Linux-pip-3.8-latest-td2752-c01906eaac36cad9507d97bc62589d8662c57aee933d68311e04bc08866eac15
Linux-pip-py3.10-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
Linux-pip-py3.10-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
Linux-pip-py3.7-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
Linux-pip-py3.7-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
Linux-pip-tdd19264-py3.10-stable-latest-6f0f1ad66a753cfc82f88e067cb37e1551d8f51ebb1fd898a7aa9c5f3475784a
Linux-pip-tdd19264-py3.10-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
Linux-pip-tdd19264-py3.10-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
Linux-pip-tdd19264-py3.7-stable-latest-6f0f1ad66a753cfc82f88e067cb37e1551d8f51ebb1fd898a7aa9c5f3475784a
Linux-pip-tdd19264-py3.7-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
Linux-pip-tdd19264-py3.7-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
Windows-pip-3.8-latest-8211468f4f94d870a0890768cc8f1e586df1cfcbcba0f18f56cbdc74a122ffa1
Windows-pip-3.8-latest-td2752-66a7f7498a08d74e669f417bc65392f27f6e444d30edad7afc034c60845d6e25
Windows-pip-3.8-latest-td2752-8211468f4f94d870a0890768cc8f1e586df1cfcbcba0f18f56cbdc74a122ffa1
Windows-pip-3.8-latest-td2752-cb6814fd3b55fdb529db8da122c4525e028a40477352fd78c900f2403c3c0fc1
Windows-pip-py3.10-stable-latest-a4f04a36ca240fbb257da322e8f735a978ed8f525cee63bd416bcdb1a238fddc
Windows-pip-py3.10-stable-latest-ea72290425a2ebd98d47b6e2be209c2a6a1b58d48ef62ccb5bf93d329a3572e3
Windows-pip-py3.7-stable-latest-a4f04a36ca240fbb257da322e8f735a978ed8f525cee63bd416bcdb1a238fddc
Windows-pip-py3.7-stable-latest-ea72290425a2ebd98d47b6e2be209c2a6a1b58d48ef62ccb5bf93d329a3572e3
Windows-pip-td-py3.10-stable-latest-a4f04a36ca240fbb257da322e8f735a978ed8f525cee63bd416bcdb1a238fddc
Windows-pip-td-py3.10-stable-latest-e8976b9aa3ee8564f6c0df352207a35e627ea95ff4c1fb9da86194c45d4501e9
Windows-pip-td-py3.10-stable-latest-ea72290425a2ebd98d47b6e2be209c2a6a1b58d48ef62ccb5bf93d329a3572e3
Windows-pip-td-py3.7-stable-latest-a4f04a36ca240fbb257da322e8f735a978ed8f525cee63bd416bcdb1a238fddc
Windows-pip-td-py3.7-stable-latest-e8976b9aa3ee8564f6c0df352207a35e627ea95ff4c1fb9da86194c45d4501e9
Windows-pip-td-py3.7-stable-latest-ea72290425a2ebd98d47b6e2be209c2a6a1b58d48ef62ccb5bf93d329a3572e3
macOS-pip-3.8-latest-912b4a7de1f6df12ab94aa2ec56c9614851f7662a0f1d8831f883e89ec11eba6
macOS-pip-3.8-latest-td2752-20d28281bf584b5b194e82c698718338e2c99ac9b6aa28e2ac6cc409b9d4c841
macOS-pip-3.8-latest-td2752-912b4a7de1f6df12ab94aa2ec56c9614851f7662a0f1d8831f883e89ec11eba6
macOS-pip-3.8-latest-td2752-c01906eaac36cad9507d97bc62589d8662c57aee933d68311e04bc08866eac15
macOS-pip-py3.10-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
macOS-pip-py3.10-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
macOS-pip-py3.7-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
macOS-pip-py3.7-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
macOS-pip-tdd19264-py3.10-stable-latest-6f0f1ad66a753cfc82f88e067cb37e1551d8f51ebb1fd898a7aa9c5f3475784a
macOS-pip-tdd19264-py3.10-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
macOS-pip-tdd19264-py3.10-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
macOS-pip-tdd19264-py3.7-stable-latest-6f0f1ad66a753cfc82f88e067cb37e1551d8f51ebb1fd898a7aa9c5f3475784a
macOS-pip-tdd19264-py3.7-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
macOS-pip-tdd19264-py3.7-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
```

</details>